src/datatypes/src/vectors/helper.rs

```diff
@@ -367,6 +371,185 @@ impl Helper {
         })
     }
 
+    /// Try to cast an vec of values into vector, fail if type is not the same across all values.
+    pub fn try_from_row_into_vector(row: Vec<Value>, dt: &ConcreteDataType) -> Result<VectorRef> {
```

Comment:
I remember that the `MutableVector` trait has a `try_push_value_ref()` method. Can we use the mutable vector to build the `VectorRef` from `Vec<Value>`?